src/display/canvas.js

```diff
@@ -52,10 +59,33 @@ function addContextCurrentTransform(ctx) {
     ctx._originalTranslate = ctx.translate;
     ctx._originalTransform = ctx.transform;
     ctx._originalSetTransform = ctx.setTransform;
+    ctx._originalResetTransform = ctx.resetTransform;
 
     ctx._transformMatrix = ctx._transformMatrix || [1, 0, 0, 1, 0, 0];
     ctx._transformStack = [];
 
+    try {
+      // The call to getOwnPropertyDescriptor throws an exception with node:
```

Comment:
Where does it do this exactly? This code is supposed to only run in Chrome, but if I call `getOwnPropertyDescriptor` in Firefox I don't get this exception, so that would mean that this is also executed in Firefox...

Reply:
> In which browser does it throw this exception exactly?

The comment mentions "node", so I'd guess that this is only required when the unit-tests are run in Node.js environments.
Perhaps the comment would be *slightly* clearer if this line was reworded to:
```suggestion
      // The call to getOwnPropertyDescriptor throws an exception in Node.js:
```

> [...] but if I call `getOwnPropertyDescriptor` in Firefox I don't get this exception, so that would mean that this is also executed in Firefox...

This *entire* block is guarded by a `if (!ctx.mozCurrentTransform) {` check above, hence none of this code should run in Firefox as far as I can tell.

Reply:
Ah, I missed the `mozCurrentTransform` guard. In that case it looks good to me. Rewording the comment would indeed have helped since I was thinking about a DOM node since it was written in lower case.